platform/inputs/analyze.go

```diff
@@ -0,0 +1,144 @@
+package inputs
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	"github.com/buildpacks/lifecycle"
+	"github.com/buildpacks/lifecycle/api"
+	"github.com/buildpacks/lifecycle/buildpack"
+	"github.com/buildpacks/lifecycle/image"
+	"github.com/buildpacks/lifecycle/internal/str"
+)
+
+// Analyze holds the values of command-line flags and args.
+// Fields are the cumulative total of inputs across all supported platform APIs.
+type Analyze struct {
+	AnalyzedPath string
+	StackPath    string
+	UID          int
+	GID          int
+	UseDaemon    bool
+
+	ForAnalyzer
+}
+
+// ForAnalyzer holds the inputs needed to construct a new lifecycle.Analyzer.
+// Fields are the cumulative total of inputs across all supported platform APIs.
+type ForAnalyzer struct {
+	AdditionalTags   str.Slice
+	CacheImageRef    string
+	LaunchCacheDir   string
+	LayersDir        string
+	LegacyCacheDir   string
+	LegacyGroupPath  string
+	OutputImageRef   string
+	PreviousImageRef string
+	RunImageRef      string
+	SkipLayers       bool
+
+	LegacyGroup buildpack.Group // for creator
+}
+
+func (a Analyze) RegistryImages() []string {
+	var images []string
+	images = appendNotEmpty(images, a.CacheImageRef)
+	if !a.UseDaemon {
+		images = appendNotEmpty(images, a.PreviousImageRef, a.RunImageRef, a.OutputImageRef)
+		images = appendNotEmpty(images, a.AdditionalTags...)
+	}
+	return images
+}
+
+type AnalyzeResolver struct {
+	PlatformAPI *api.Version
+}
+
+// Resolve accepts AnalyzeInputs with flags filled in, and args.
+// It returns AnalyzeInputs with default values filled in, or an error if the provided inputs are not valid.
```

Comment:
```suggestion
// Resolve accepts Analyze with flags filled in, and args.
// It returns Analyze with default values filled in, or an error if the provided inputs are not valid.
```
```suggestion
// Resolve accepts AnalyzeInputs with flags filled in, and args.
// It returns AnalyzeInputs with default values filled in, or an error if the provided inputs are not valid.
```

Reply:
IDK about this struct name

Reply:
IDK about this struct name

Reply:
I like how `Inputs` alludes to the spec.